### PR TITLE
Replace with Carthage based setup and eliminate Specta from tests

### PR DESCRIPTION
**Pull request by meaning-matters**
This reorganizes files, and replaces CocoaPods with Carthage.

Under CocoaPods all cloned pods sources were part of this repo; for Carthage I've chosen, as discussed, to not include cloned repos. But Carthage has the same option, so if we think it's better to include everything, we can easily do that.

I had to rename the `UPTEthereumSigner` class to `UPTEthSigner` because the `UPTEthereumSigner.h` must be used as framework API/header.

Specta, an old Obj-C testing package has been eliminated; iOS' own testing framework is sufficient and the code looks much better without Specta. There are of lot of tests failing; I guess it's because these tests are outdated, but I still need to check if that's 'correct'.

I've also reformatted the existing code (quite some lines were very long, had inconsistent spacing, had unclear indentation, ...).  Looks much better now and in line with recently added JWT verification, DID resolving, and Sodium encryption work.  (I thought it was a good time for that because no logic code changes/additions were made; but I could undo that for now and place it in a separate PR.)

### CODE REVIEW

**Comment by meaning-matters** [comment]
Xcode project and config files may still change a bit. Changes might be needed when trying to use this framework in the uPort SDK.

**Comment by meaning-matters** [comment]
Tests only run on the simulator at the moment. A test app needs to be added to allow them to run on a device; I want to add that consistently to all new Carthage/framework-based repos later.

File: UPTEthereumSigner/UPTEthSigner.h [hunk 1]
```diff
@@ -30,23 +27,39 @@ typedef void (^UPTEthSignerDeleteKeyResult)(BOOL deleted, NSError *error);
 
 @class VALValet;
 
-@interface UPTEthereumSigner : NSObject
+@interface UPTEthSigner : NSObject
```
Comment (mirceanis) [inline UPTEthereumSigner/UPTEthSigner.h:30]: The `UPTEthereumSigner` class is part of the main API of this library and renaming it seems like an unnecessary breaking change that will force downstream developers to refactor.
Is there a reason for renaming it that I'm not aware of?

**Review by mirceanis — CHANGES_REQUESTED** [review]
I'm really glad so much code has been removed from this repo.
The `UPTEthereumSigner`->`UPTEthSigner` rename seems like an unnecessary breaking change so I'm requesting that be undone.
Otherwise nice work, thanks!

**Comment by meaning-matters** [comment]
Done

**Pull request closed without merging**